test/routes/testSoldierRoutes.js

```diff
@@ -0,0 +1,98 @@
+
+describe("Soldiers tests", function () {
+
+  before("Initialaizing db", doneBefore => {
+    MongoClient.connect(DBurl, function (dbConnectErr, client) {
+      if (dbConnectErr) return doneBefore(dbConnectErr);
+      testServer.dbClient = client;
+      testServer.collection = client.db(dbName).collection('Soldiers')
+      doneBefore();
+    });
+    
+  });
+
+  afterEach("db cleaning", doneAfterEach => {
+    testServer.collection.deleteMany({}, (deleteError) => {
+      if(deleteError) return doneAfterEach(deleteError);
+      doneAfterEach();
+    })
+  });
+
+  after("Closing the server", doneAfter => {
+    testServer.close();
+    testServer.dbClient.close();
+    doneAfter();
+  });
+
+  function testsAssistant(httpMethod, expectedResults, testData, httpStatus, testFunc, urlAddon, soldierToSearch, tadone) {
+    const Http = new XMLHttpRequest();
+    Http.open(httpMethod, url + urlAddon);
+    Http.send(testData);
+    Http.onreadystatechange = (connectErr) => {
+      if(connectErr) return tadone(connectErr);
+      if (Http.readyState == 4 && Http.status == httpStatus) {
+        if (testFunc) {
+          testFunc(soldierToSearch, (testFuncErr, res) => {
+            if(testFuncErr) return tadone(testFuncErr)
+            if(res) expectedResults = res
+            expect(Http.responseText).to.eql(expectedResults);
+            return tadone();
+          })
+        } else {
+          expect(Http.responseText).to.eql(expectedResults);
+          return tadone();
+        }
+      }
+    }
+  }
+
+  describe("Soldiers post test", function () {
+
+    it("Should return an error message when the soldier data is missing fields", function (done) {
+      httpMethod = "POST", expectedResults = "One or more fields are invalid"
+      testData = JSON.stringify({
+        "id": solId,
+        "name": solName,
+        "limitations": limitations
+      })
+      httpStatus = 404, testFunc = null, urlAddon = "", solToSearch = {}
+      testsAssistant(httpMethod, expectedResults, testData, httpStatus, testFunc, urlAddon, solToSearch, done)
+    })
+
+    it("Should return an empty response when the soldier has been added to the database", function (done) {
+      httpMethod = "POST", expectedResults = '', testData = standartSol
+      httpStatus = 200, testFunc = null, urlAddon = "", solToSearch = {}
+      testsAssistant(httpMethod, expectedResults, testData, httpStatus, testFunc, urlAddon, solToSearch, done)
+    })
+
+    it("Should return error message when the soldier data has extra fields", function (done) {
+      httpMethod = "POST", expectedResults = "One or more fields are invalid",
+        testData = JSON.stringify({
+          "id": solId,
+          "name": solName,
+          "stam": "lol",
+          "limitations": limitations
+        })
+      httpStatus = 404, testFunc = null, urlAddon = "", solToSearch = {}
+      testsAssistant(httpMethod, expectedResults, testData, httpStatus, testFunc, urlAddon, solToSearch, done)
```

Comment:
Maybe Pavel talked with you about this,
usually too many arguments is confusing because you have to remember them by order, and if later you add some more you'll certainly make a bug somewhere.

So arguments that have a common ground should be combined.
Usually - all arguements except the callback
`testsAssistant({ httpMethod, expectedResults, ..., solToSearch }, done)`